model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java

```diff
@@ -62,6 +62,12 @@ public void update(T entity) {
 
     public void delete(T entity) {
         if (entity != null) {
+            // making sure the entity in question,
+            // is really part of this transaction
+            if (! entityManager.contains(entity)) {
+                entity = entityManager.merge(entity);
```

Comment:
that's also conficting w/ changes done from Lukas on the JMS.

I will file a JIRA for future evaluation of improving this

On Wed, Apr 15, 2015 at 2:57 PM, Erik Jan de Wit <notifications@github.com>
wrote:

> In
> model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPABaseDao.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/529#discussion_r28416322>
> :
>
> > @@ -62,6 +62,12 @@ public void update(T entity) {
> >
> >      public void delete(T entity) {
> >          if (entity != null) {
> > +            // making sure the entity in question,
> > +            // is really part of this transaction
> > +            if (! entityManager.contains(entity)) {
> > +                entity = entityManager.merge(entity);
>
> sure @matzew <https://github.com/matzew> guess we only need to say T
> extends BaseEntity or something like that, but let's do that later
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/529/files#r28416322>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf